plugins/ltac/taccoerce.ml

```diff
@@ -253,7 +250,7 @@ let id_of_name = function
     | Some c ->
        match EConstr.kind sigma c with
        | Var id -> id
-       | Meta m -> id_of_name (Evd.meta_name sigma m)
+       | Meta m -> Id.of_string "x"
```

Comment:
>Metas should not escape internals of unification and clenv, so Ltac should never
encounter a meta at runtime when inspecting terms.

why isn't this assert false then?

Reply:
It doesn't hurt to be more lenient. The goal really is to get rid of meta_name.